helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -548,14 +561,14 @@ public void addConfigChangeListener(org.apache.helix.ScopedConfigChangeListener
   @Override
   public void addMessageListener(MessageListener listener, String instanceName) {
     addListener(listener, new Builder(_clusterName).messages(instanceName), ChangeType.MESSAGE,
-        new EventType[] { EventType.NodeChildrenChanged });
+        new EventType[] { EventType.NodeChildrenChanged }, _messageRefreshTriggerInterval);
   }
 
   @Deprecated
   @Override
   public void addMessageListener(org.apache.helix.MessageListener listener, String instanceName) {
     addListener(listener, new Builder(_clusterName).messages(instanceName), ChangeType.MESSAGE,
-        new EventType[] { EventType.NodeChildrenChanged });
+        new EventType[] { EventType.NodeChildrenChanged }, _messageRefreshTriggerInterval);
```

Comment:
optional, but I guess we don't need to add support for the deprecated methods.

And one question, why not just casting the deprecated listener class to the newer one? So it becomes maintenance-free.

Reply:
Good idea. This is better way to do for the deprecated listener. Updated it.